server/controllers/CommentController.js

```diff
@@ -0,0 +1,145 @@
+import db from '../models';
+import response from '../helpers/response';
+import TokenManager from '../helpers/TokenManager';
+
+
+const { Article, Comment } = db;
+
+/**
+ * @class UserController
+ */
+class CommentController {
+  /**
+   *
+   * @description Allow user to comment on an article
+   * @static
+   * @param {*} req Express Request object
+   * @param {*} res Express Response object
+   * @returns {object} Json response
+   * @memberof CommentController
+   */
+  static async addComment(req, res) {
+    try {
+      const token = req.headers.authorization.split(' ')[1];
+      const { userId } = TokenManager.verify(token);
+      const { content } = req.body;
+      const { articleId } = req.params;
+      if (!content) {
+        return response(res, 400, 'failure', 'Enter a comment', null, null);
+      }
+      await Article.findAndCountAll({
```

Comment:
Nice work on the logic TJ. 
I'm lost on what is happening on line 30 to 34.
Why are you finding and counting which fires two queries to the db? 
Isn't that rather expensive? 
I reckon `Article.findOne()` should do a decent job.
Also, how about you store the result of the operation on the aforementioned line into a variable, so you can check if the article that is about to be commented on exists. That check hasn't been done.

Reply:
Thanks will infect the change on Article.findOne() but that of checking if the article that is about to be commented on exists error is handled by the catch block.

Reply:
The catch block is going to be called when an error is THROWN. How does the logic perform this throw and fire the catch block? At what point is this happening?

Reply:
yes true thanks have fix that